lua/ttt2/libraries/weaponrenderer.lua

```diff
@@ -353,3 +361,50 @@ function weaponrenderer.Render(wep, elements, boneEntity)
         end
     end
 end
+
+---
+-- Renders the view model if valid view model elements are provided. It also updates the bone
+-- positions so that the view model tracks the hands.
+-- @param Entity wep The weapon whose view model should be rendered
+-- @param table elements The elements of the view model
+-- @param Entity viewModel The player's view model
+-- @realm client
+function weaponrenderer.RenderViewModel(wep, elements, viewModel)
+    if not elements then
+        return
+    end
+
+    weaponrenderer.UpdateBonePositions(wep, viewModel)
+
+    weaponrenderer.Render(wep, elements, viewModel)
+end
+
+---
+-- Renders the wold model if valid wold model elements are provided. It also renders the default
+-- world model of the weapon if enabled
+-- @param Entity wep The weapon whose world model should be rendered
+-- @param Entity wepModel The weapon mode whose world model should be rendered, in most cases
+-- identical to the first parameter
+-- @param table elements The elements of the world model
+-- @param[opt] Player owner The owner entity of the weapon, binds the model to their hands
+-- @realm client
+function weaponrenderer.RenderWoldModel(wep, wepModel, elements, owner)
+    if wep.ShowDefaultWorldModel then
+        wepModel:DrawModel()
+    end
+
+    if not elements then
+        return
+    end
```

Comment:
same here

Reply:
Not sure how to do it here, hm.
This function also draws the normal weapon world model and the custom one can only be drawn if it is set up. Therefore an external check doesn't work.

I can change it for the view model, but that makes the handling between these two functions incosistent